corehq/apps/users/models.py

```diff
@@ -2951,7 +2951,9 @@ class Meta(object):
 
 class ApiKeyManager(models.Manager):
     def get_queryset(self):
-        return super().get_queryset().filter(is_active=True)
+        return super().get_queryset()\
+            .filter(is_active=True)\
+            .exclude(expiration_date__lt=datetime.now())
```

Comment:
I had a brief moment of panic that this would exclude keys where the expiration date is unspecified (as `.filter(expiration_date__gt=datetime.now())` would), but it doesn't, since it's excludes on a positive match, rather than including on a positive match.  Do you know if there are any tests that specifically verify that?

Reply:
(I also had that moment of panic.)

Reply:
@esoergel There are two tests for api key auth without the expiration date set. One with an active and one with an inactive key

Reply:
thanks for confirming!